File: backend/src/main/kotlin/com/skiploom/domain/Recipe.kt
package com.skiploom.domain

import java.util.UUID

data class Ingredient(
    val id: UUID,
    val amount: Double,
    val unit: String,
    val name: String
)

data class Step(
    val id: UUID,
    val orderIndex: Int,
    val instruction: String
)

data class Recipe(
    val id: UUID,
    val title: String,
    val description: String?,
    val ingredients: List<Ingredient>,
    val steps: List<Step>
)

File: frontend/src/types/recipe.ts
export interface Ingredient {
  id: string;
  amount: number;
  unit: string;
  name: string;
}

export interface Step {
  id: string;
  orderIndex: number;
  instruction: string;
}

export interface Recipe {
  id: string;
  title: string;
  description?: string;
  ingredients: Ingredient[];
  steps: Step[];
}
